### ISSUE TITLE
Admin\Operations - External user can edit an Operation

### ISSUE THREAD

**Issue by andrea-williams** [bcgov/cas-registration#1879]
#### Dependencies:
- Blocked by #1694 
- 
- #### Description:

"Edit" modes of Operation Information page

#### Acceptance Criteria:

Given I am an authenticated external user
When I click on the Operation Information page and click the 'Edit' button
Then I am able to edit the data fields relevant to my Operation

Edit:
[Figma wireframe](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=18025-1336&m=dev)

#### Development Checklist:

- [ ] don't sweat over the green checkmark if data validation from live editing is too computationally expensive or dev-intensive
- [ ] edited data that is 'submitted' propagates to the database
- [ ] Checklist item
- [ ] Meets the DOD

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

·**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

**Comment by patriciarussellCAS** [comment]
Review notes: 
On the Operations table page I'm a bit confused by the the difference between actions in the Facilities column and the Actions column. 
For example: For Operation 16, under Facilities it says "Edit details" which when clicked, takes you to the facility page in edit mode. But under Action, you click "View Operation" which takes you to the Operation Information page in view only mode. Then on the Operation Information page you click on "edit" if you want to make changes. 
I'm not clear why the steps are different between facility and operation. I think my expectation here was that it would say "view facility" in a similar way and then on the facility information page they click "Edit" if they need to change something. Would love some more context on this!

**Comment by marcellmueller** [comment]
@patriciarussellCAS I think that's fair though that wasn't part of this ticket. The Operation DataGrid is implemented as the wireframes are shown, so maybe something to bring up with the designers.

I'd also note that the Facility DataGrid shows `View Facility` for an SFO Operation if that single Facility information has been saved. It shows `Edit Details` if there is no Facility saved for that SFO Operation. I forget the details but I think this might be for Operations imported through swrs? So I think it's intentional to have `View Operation` for Operations that are Registered, and `Start Registration/Continue Registration` is similar to the idea behind `Edit Details`.

Though the other day I noticed the `Start Registration` and `Continue Registration` links in the DataGrid should probably take you to the Registration form, not the Operation details form and created a ticket:

https://github.com/bcgov/cas-registration/issues/2270

**Comment by patriciarussellCAS** [comment]
Thanks Marcel - I agree it's a bit unrelated to this specific ticket. Just wanted to note it before I forgot. Thanks for creating a follow-up ticket ;)
